sale_purchase_according_declared_stock/models/purchase_order.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    # TODO: Do we also want to add the picking status to the purchase order tree view?
+    # TODO check if you have to do something different than what is done in sale_purchase when the purchase is canceled
+    def button_confirm(self):
+        result = super().button_confirm()
+        self._create_detailed_operations()
+        return result
+
+    def _create_detailed_operations(self):
+        num_pickings = 0
+        pick_ids = self.picking_ids.filtered(
+            lambda l: l.state != 'cancel')
+        for p in pick_ids:
+            if p.state == 'assigned':
+                num_pickings += 1
+        if num_pickings > 0:
```

Comment:
Suggested minimal code:
```suggestion
        pick_ids = self.picking_ids.filtered(
            lambda l: l.state != 'cancel')
        if pick_ids.filtered(lambda l: l.state == 'assigned'):
```

Reply:
You're right. Now it seems obvious ... the issue is how I started and what I put the filter in afterwards ...